### PR TITLE
[PFX-726] - Overhaul docs & generated app docs

### PR DESCRIPTION
**Pull request by mmason2-godaddy**
## Summary
Overhaul generated documentation in Gasket apps for provide more value and information to the end user.

- [Root TypeScript Docs](https://github.com/godaddy/gasket/blob/PFX-726/docs/typescript.md) 📖 
- [ESM + TypeScript](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-typescript/docs/esm-typescript.md) 📖 
- [Preset NextJS Generated README](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-preset-nextjs/generator/README.md) 📖 
- [Preset API Generated README](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-preset-api/generator/README.md) 📖 
- [Readme API](https://github.com/godaddy/gasket/blob/PFX-726/packages/create-gasket-app/lib/scaffold/readme.js#L4)
- [Add Handlebars helper to compile nested HB markup](https://github.com/godaddy/gasket/blob/PFX-726/packages/create-gasket-app/lib/scaffold/actions/generate-files.js#L113)

#### Markdown Partials
- `@gasket/plugin-nextjs`
  - [App Router](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-nextjs/generator/markdown/app-router.md) 📖 
  - [Page Router](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-nextjs/generator/markdown/pages-router.md) 📖 
  - [Custom Server](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-nextjs/generator/markdown/custom-server.md) 📖 
- `@gasket/plugin-typescript`
  - [Partial](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-typescript/generator/markdown/README.md) 📖 

#### Readme API Implemetation
- [Docs plugin](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-docs/lib/create.js#L31)
- [Docusaurus plugin](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-docusaurus/lib/create.js#L20)
- [NextJS plugin](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-nextjs/lib/create.js#L27)
- [Swagger plugin](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-swagger/lib/index.js#L139)
- [TypeScript plugin](https://github.com/godaddy/gasket/blob/PFX-726/packages/gasket-plugin-typescript/lib/create.js#L24)

## Changelog
- Update generated docs in Gasket apps
- Add Readme API

## Test Plan
Tests & types updated

### CODE REVIEW

File: docs/typescript.md [hunk 1]
```diff
@@ -2,89 +2,70 @@
...
 
-## Validating gasket.config.js
+## Writing and Validating Gasket Configuration
 
-The `@gasket/engine` package supplies a `GasketConfigFile` type that, once you've also imported all your plugins, validates the contents of your gasket config file.
+In Gasket, you can now write your `gasket.js` configuration file as `gasket.ts` using TypeScript. This allows for full type-checking and validation directly within your TypeScript environment.
 
-The `gasket.config.js` file cannot be written in TypeScript, but you can configure TypeScript to check JS files or use a `@ts-check` comment along with JSDoc type annotations. Due to JSDoc type checking limitations, you may have to separate the config declaration and your export:
+The `@gasket/core` package supplies a `GasketConfigFile` type that validates the contents of your Gasket config file. When you import all your plugins into `gasket.ts`, TypeScript can fully validate the configuration.
```
Comment (agerard-godaddy) [inline docs/typescript.md:13]:
```suggestion
The `@gasket/core` package supplies a `GasketConfigDefinition` type that validates the contents of your Gasket config file. When you import all your plugins into `gasket.ts`, TypeScript can fully validate the configuration.
```

File: docs/typescript.md [hunk 2]
```diff
@@ -2,89 +2,70 @@
...
+import pluginA from '@gasket/plugin-a';
+import pluginB from '@gasket/plugin-b';
+
+export default makeGasket({
+  plugins: [
+    pluginA,
+    pluginB
+  ]
+  // additional config
+} as GasketConfigDefinition);
```
Comment (agerard-godaddy) [inline docs/typescript.md:30]: You shouldn't have to explicitly cast the config when configuring in the `makeGasket` argument.

File: docs/typescript.md [hunk 3]
```diff
@@ -2,89 +2,70 @@
...
 
-## Ensuring visibility of plugin extensions
+## Ensuring Visibility of Plugin Extensions
 
-Because Gasket itself is just a plugin framework, plugins themselves are responsible for augmenting the core Gasket interfaces. This means that the type declarations for the plugins have to be "discovered" in your TypeScript code base. This can be done by ensuring you reference or import your presets or plugins. The sample code snippets below show these imports, but if you can ensure TypeScript knows about all of the plugins from one centralized area of code then you can avoid duplication.
+Because Gasket is a plugin framework, plugins themselves are responsible for augmenting the core Gasket interfaces. This means that the type declarations for the plugins must be "discovered" in your TypeScript codebase. You can ensure this by referencing or importing your presets or plugins. The sample code snippets below show these imports, but if you can centralize the imports in one area of your codebase, you can avoid duplication and ensure that TypeScript is aware of all the plugins.
 
-## Validating gasket.config.js
+## Writing and Validating Gasket Configuration
```
Comment (agerard-godaddy) [inline docs/typescript.md:9]: A bit verbose for a heading - how about
```suggestion
## Validating Gasket Configuration
```

File: docs/typescript.md [hunk 4]
```diff
@@ -2,89 +2,70 @@
...
-/** 
- * @typedef {import('@gasket/engine').Hook<'intlLocale'>} IntlLocaleHandler
- */
+```typescript
+import type { HookHandler } from '@gasket/core';
 
-/** @type {IntlLocaleHandler} */
-const handler = (gasket, currentLocale, { req, res }) => {
-  // return 3; - does not pass validation
+const intlLocaleHandler: HookHandler<'initLocale'> = (gasket, locale, { req, res }) => {
```
Comment (agerard-godaddy) [inline docs/typescript.md:40]: Is the `///<reference types="@gasket/plugin-intl"/>` still necessary here?

File: packages/gasket-plugin-typescript/docs/esm-typescript.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+```ts
+import { myFunction } from './myModule';
+```
+
+When switching to ESM, TypeScript does not automatically append file extensions, leading to runtime errors if the correct files are not found. To comply with ESM's requirements, developers must include the file extensions explicitly:
+
+```ts
+import { myFunction } from './myModule.js';
+```
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-typescript/docs/esm-typescript.md:25]: The rendered markdown does not surface if this is ts or js. Can you add comments to these blocks and the ones below?

> ```diff
> + //example.js
> import { myFunction } from './myModule.js';
> ```
> 
> In contrast, in TypeScript (when using CommonJS):
> 
> ```diff
> + // example.ts
> import { myFunction } from './myModule';
> ```
> 
> When switching to ESM, TypeScript does not automatically append file extensions, leading to runtime errors if the correct files are not found. To comply with ESM's requirements, developers must include the file extensions explicitly:
> 
> ```diff
> + // example.ts
> import { myFunction } from './myModule.js';
> ```

File: docs/typescript.md (hunk 4 above)
Comment (mmason2-godaddy) [reply docs/typescript.md:40]: Maybe? I was testing with defining hook handlers and I didn't see an issues 🤷‍♂️

File: docs/typescript.md (hunk 2 above)
Comment (mmason2-godaddy) [reply docs/typescript.md:30]: Removed & removed from the `write-gasket-config`

**Review by agerard-godaddy — APPROVED** [review]
🥇

**Review by jpina1-godaddy — APPROVED** [review]

**Pull request merged**